### PR TITLE
Added the option for image to take its original size

### PR DESCRIPTION
**Pull request by MisustinIvan**
you can now do .Size(0,0) and the image takes its original size

### CODE REVIEW

**Comment by MisustinIvan** [comment]
Because i am not if there is a way to do this without this change if you use image from URL. You can of course just write your own function that fetches the image and gets the size, but whats the point of ImageWithURLWidget? I was also thinking about adding a option to set a size constraint and just scale the image while keeping the aspect ratio to fit the constraint. It would be nice if you shared some feedback, thanks!

**Comment by MisustinIvan** [comment]
I have also thought about replacing the behavior when passing in (-1, -1) to Size with this, because the current one kinda sucks and just stretches the Image.

File: ImageWidgets.go [hunk 1]
```diff
@@ -292,6 +294,14 @@ func (i *ImageWithURLWidget) Timeout(downloadTimeout time.Duration) *ImageWithUR
 
 // Size sets image's size.
 func (i *ImageWithURLWidget) Size(width, height float32) *ImageWithURLWidget {
+
+	if width == 0 && height == 0 {
```
[the remark this answers is not in the document]
Comment (MisustinIvan) [reply ImageWidgets.go:298]: sure, will fix that!

File: ImageWidgets.go [hunk 2]
```diff
@@ -292,6 +294,14 @@ func (i *ImageWithURLWidget) Timeout(downloadTimeout time.Duration) *ImageWithUR
 
 // Size sets image's size.
 func (i *ImageWithURLWidget) Size(width, height float32) *ImageWithURLWidget {
+
```
[the remark this answers is not in the document]
Comment (MisustinIvan) [reply ImageWidgets.go:297]: is there a way for me to see CI runs in my fork? Do i just create my own workflows?

**Comment by MisustinIvan** [comment]
And would you be able to recommend me some things to work on in this project? I find it quite cool and would like to contribute some more. :smile:

File: ImageWidgets.go [hunk 3]
```diff
@@ -74,14 +74,24 @@ func (i *ImageWidget) Size(width, height float32) *ImageWidget {
 
 // Build implements Widget interface.
 func (i *ImageWidget) Build() {
+
```
Comment (gucio321) [inline ImageWidgets.go:77]: linter still complains here

**Review by gucio321 — CHANGES_REQUESTED** [review]
2 more things:
- could you wite a comment over ImageWidget (and probably ImageWithURL and other Image*` type definition and describe what `0` does
- there is still one liner issue

**Comment by gucio321** [comment]
@MisustinIvan generally, we have 3 types of issues:
1. giu-related stuff - e.g. missing functions, some nice enhancements for API (or things like this PR). I've tagged them with https://github.com/AllenDang/giu/labels/good%20first%20issue
2. issues with cimgui-go (which are the worst sh*t I've ever seen :smile:). In general they are caused by our migration from imgui-go to cimgui-go and the fact that cimgui-go was/is early in development. It works but it requires some fixes.
3. well, we have one more _type of issue_: since migration in #628 I was forced to disable some widgets (e.g. CodeEditor, MarkdownEditor end so on). Now it'd be nice to re-enable them but it requires work on cimgui-go and even writing some C wrapers for C++ projects in lua (basing on https://github.com/cimgui like I did e.g. for markdown in https://github.com/gucio321/cimmarkdown)

**Comment by MisustinIvan** [comment]
Bruh does the linter really want comments to end with period...

**Comment by MisustinIvan** [comment]
Check passed, everything seems fine to me!

**Review by gucio321 — APPROVED** [review]
nice, thank you

**Comment by gucio321** [comment]
> Bruh does the linter really want comments to end with period...

you can use `golangci-lint run --fix ./...` and this should automatically fix that.

File: ImageWidgets.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (gucio321) [reply ImageWidgets.go]: you should see them here:
image

**Comment by MisustinIvan** [comment]
thanks, will try that next time!

**Comment by MisustinIvan** [comment]
btw what is the best practice, github is asking me if I want to delete the feature branch. Should i delete it or keep it?

**Comment by gucio321** [comment]
I thing you should delete. these commits are in master anyway and if you keep every branch, you'll have hundreds of them some day :smile:

**Pull request merged**